`key.properties` is required even when building in debug mode

## Description
When you clone this repository and run `flutter run` for an Android device, you will get this error:
```
FAILURE: Build failed with an exception.

* Where:
Build file '/Users/nils/Desktop/Projects/sharezone/test/sharezone-app/app/android/app/build.gradle' line: 21

* What went wrong:
A problem occurred evaluating project ':app'.
> /Users/nils/Desktop/Projects/sharezone/test/sharezone-app/app/android/key.properties (No such file or directory)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
```

This is because the `app/build.gradle` tries to load the `key.properties` for Android code signing every time - no matter if you're building in release or debug mode.

https://github.com/SharezoneApp/sharezone-app/blob/8beeb01620eef3011668d3020e765f3857555fa3/app/android/app/build.gradle#L21

## Solution
We should only the load the `key.properties` when you really build in release mode.

--- a/app/android/app/build.gradle
+++ b/app/android/app/build.gradle
@@ -18,7 +18,8 @@ apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 def keystorePropertiesFile = rootProject.file("key.properties")
 def keystoreProperties = new Properties()
-if (keystorePropertiesFile.exists()) {
+def hasKeystorePropertiesFile = keystorePropertiesFile.exists()
+if (hasKeystorePropertiesFile) {
     keystoreProperties.load(new FileInputStream(keystorePropertiesFile))
 }
 
@@ -52,7 +53,7 @@ android {
     }
 
     signingConfigs {
-        if(keystorePropertiesFile.exists()) {
+        if(hasKeystorePropertiesFile) {
             release {
                 keyAlias keystoreProperties['keyAlias']
                 keyPassword keystoreProperties['keyPassword']
@@ -63,7 +64,7 @@ android {
     }
     buildTypes {
         release {
-            if(keystorePropertiesFile.exists()) {
+            if(hasKeystorePropertiesFile) {
                 signingConfig signingConfigs.release
             }
 

--- a/app/android/app/build.gradle
+++ b/app/android/app/build.gradle
@@ -18,7 +18,9 @@ apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 def keystorePropertiesFile = rootProject.file("key.properties")
 def keystoreProperties = new Properties()
-keystoreProperties.load(new FileInputStream(keystorePropertiesFile))
+if (keystorePropertiesFile.exists()) {
+    keystoreProperties.load(new FileInputStream(keystorePropertiesFile))
+}
 
 def flutterVersionCode = localProperties.getProperty('flutter.versionCode')
 if (flutterVersionCode == null) {
@@ -50,16 +52,21 @@ android {
     }
 
     signingConfigs {
-        release {
-            keyAlias keystoreProperties['keyAlias']
-            keyPassword keystoreProperties['keyPassword']
-            storeFile file(keystoreProperties['storeFile'])
-            storePassword keystoreProperties['storePassword']
+        if(keystorePropertiesFile.exists()) {
+            release {
+                keyAlias keystoreProperties['keyAlias']
+                keyPassword keystoreProperties['keyPassword']
+                storeFile file(keystoreProperties['storeFile'])
+                storePassword keystoreProperties['storePassword']
+            }
         }
     }
     buildTypes {
         release {
-            signingConfig signingConfigs.release
+            if(keystorePropertiesFile.exists()) {
+                signingConfig signingConfigs.release
+            }
+
             minifyEnabled true
             // Proguard is needed to run Jitsi in release mdoe
             useProguard true

